modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/resource/v1_0/MasterPageResourceImpl.java

```
@@ -380,6 +405,39 @@ else if (_isPublishedLayout(masterPage.getPageSpecifications())) {
 				serviceContext));
 	}
 
+	private String[] _getAssetTagNames(
+			long groupId, ItemExternalReference[] itemExternalReferences)
+		throws Exception {
+
+		if (ArrayUtil.isEmpty(itemExternalReferences)) {
+			return new String[0];
+		}
+
+		String[] assetTagNames = new String[itemExternalReferences.length];
+
+		Group group = _groupService.getGroup(groupId);
+
+		for (int i = 0; i < itemExternalReferences.length; i++) {
+			long scopeGroupId = groupId;
+
+			if (Validator.isNotNull(itemExternalReferences[i].getScope())) {
+				scopeGroupId = GroupUtil.getGroupId(
+					true, true, group.getCompanyId(),
+					itemExternalReferences[i].getScope(
+					).getExternalReferenceCode());
+			}
+
+			AssetTag assetTag =
```

Shall we throw UnsupportedOperationException here to mark that a possible missing reference will need to be lazily handled?

---

If assetTag is null we throw a UnsupportedOperationException